src/inline-svg.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, Renderer2, RendererFactory2 } from '@angular/core';
+import { InlineSVGDirective } from './inline-svg.directive';
+
+@Injectable()
+export class InlineSVGService {
+    private _renderer: Renderer2;
+
+    constructor(rendererFactory: RendererFactory2) {
+        this._renderer = rendererFactory.createRenderer(null, null);
+    }
+
+    insertEl(
+        dir: InlineSVGDirective,
+        parentEl: HTMLElement,
+        content: Element,
+        replaceContents: boolean,
+        prepend: boolean) {
+        if (replaceContents && !prepend) {
+            const parentNode = dir._prevSVG && dir._prevSVG.parentNode;
+            if (parentNode) {
+                this._renderer.removeChild(parentNode, dir._prevSVG);
+            }
+
+            parentEl.innerHTML = '';
```

Comment:
Is there a more "Renderer"-y way of doing this? Not sure if directly manipulating the elements is considered an antipattern.

Reply:
I wondered this myself during refactoring, but couldn't find an alternative.
I can try to replace it with the `createText` function of renderer. This might work at least for every simple case.

Reply:
So long as it works, it's fine. I expect things to change again once Renderer3 is out anyway.